src/storage_credential.cpp

```diff
@@ -34,15 +34,20 @@ namespace azure {  namespace storage_lite {
         string_to_sign.append("\n"); // Range
 
         // Canonicalized headers
-        for (const auto &header : headers.ms_headers)
+        std::map<std::string, std::string> ordered_ms_headers;
```

Comment:
>map [](start = 13, length = 3)

Is it compulsory for the header to be alphabetically ordered?

Reply:
Yes, in this [document](https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key). `3. Sort the headers lexicographically by header name, in ascending order`